lib/animina/accounts/resources/user.ex

```diff
@@ -503,6 +496,12 @@ defmodule Animina.Accounts.User do
     attribute :minimum_partner_height, :integer, allow_nil?: true
     attribute :maximum_partner_height, :integer, allow_nil?: true
 
+    attribute :confirmation_pin, :integer do
```

Comment:
This shouldn't be able to be nil. The PIN or better the hash should be created during the create action. If I create a new user in the iex I want it to have a PIN/hash automatically.

Actually the `confirmation_pin` should be a virtual field which can be written to when I create a user in the iex. That input should be automatically converted to the hash and stored in the hash.

Reply:
@wintermeyer  hello , if we hash with bcrypt and store it a pin that is "123456" as "$2b$12$uf/Ow4LXf54uGjo5A3WrP.aWOKDEHQu3zCVbFBgwKmQ8fbg837xLm" , knowing there is no way to get the value from a hash , we can only compare 2 values , how would we send the email to the user with this pin , keep in mind too , on that page they can request for another email to be sent , so we have to store that value explicitly some way , thoughts?

Reply:
We write a `set_new_pin_and_send_by_email` function which does exactly that. It generates a new PIN, stores the hash, resets the counter and sends the email to the user. The PIN is only known within that function. When a user requests a second email he gets a new PIN.

BTW: We want to include the UserID in the hash too. Otherwise it would be easy to compare to users who have the same PIN.

Reply:
The `set_new_pin_and_send_by_email` function is called in the `create` action.

Reply:
Okay , that does make sense , I have set it as such  we have this in the before create action , so we cannot access ther user_id in this case